src/theory/sets/solver_state.h

```diff
@@ -0,0 +1,290 @@
+  eq::EqualityEngine& d_ee;
+  /** The list of all equivalence classes of type set in the current context */
+  std::vector<Node> d_set_eqc;
+  /** Maps types to the equivalence class containing empty set of that type */
+  std::map<TypeNode, Node> d_eqc_emptyset;
+  /** Maps types to the equivalence class containing univ set of that type */
+  std::map<TypeNode, Node> d_eqc_univset;
+  /** Maps equivalence classes to a singleton set that exists in it. */
+  std::map<Node, Node> d_eqc_singleton;
+  /** Map from terms to the representative of their congruence class */
+  std::map<Node, Node> d_congruent;
+  /** Map from equivalence classes to the list of non-variable sets in it */
+  std::map<Node, std::vector<Node> > d_nvar_sets;
+  /** Map from equivalence classes to a variable sets in it */
+  std::map<Node, Node> d_var_set;
+  /** polarity memberships
+   *
+   * d_pol_mems[0] maps equivalence class to their positive membership list
+   * with explanations (see getMembers), d_pol_mems[1] maps equivalence classes
+   * to their negative memberships.
+   */
+  std::map<Node, std::map<Node, Node> > d_pol_mems[2];
+  // --------------------------------------- commonly used terms
+  /** Map from set terms to their proxy variables */
+  NodeMap d_proxy;
+  /** Backwards map of above */
+  NodeMap d_proxy_to_term;
+  /** Cache of type constraint skolems (see getTypeConstraintSkolem) */
+  std::map<Node, std::map<TypeNode, Node> > d_tc_skolem;
+  /** Map from types to empty set of that type */
+  std::map<TypeNode, Node> d_emptyset;
+  /** Map from types to universe set of that type */
+  std::map<TypeNode, Node> d_univset;
+  // --------------------------------------- end commonly used terms
+  // -------------------------------- term indices
+  /** Term index for MEMBER
+   *
+   * A term index maps equivalence class representatives to the representative
+   * of their congruence class.
+   *
+   * For example, the term index for member may contain an entry
+   * [ r1 -> r2 -> (member t1 t2) ] where r1 and r2 are representatives of their
+   * equivalence classes, (member t1 t2) is the representative of its congruence
+   * class, and r1=t1 and r2=t2 hold in the current context.
+   */
+  std::map<Node, std::map<Node, Node> > d_members_index;
+  /** Term index for SINGLETON */
+  std::map<Node, Node> d_singleton_index;
+  /** Indices for the binary kinds INTERSECT, SETMINUS and UNION. */
+  std::map<Kind, std::map<Node, std::map<Node, Node> > > d_bop_index;
+  // -------------------------------- end term indices
+  std::map<Kind, std::vector<Node> > d_op_list;
+  /** the skolem cache */
+  SkolemCache d_skCache;
+  /** is set disequality entailed internal
+   *
+   * This returns true if disequality between sets a and b is entailed in the
+   * current context. We use an incomplete test based on equality and membership
+   * information.
+   *
+   * eqE is the representative of the equivalence class of the empty set
```

Comment:
`eqE` is the representative of .... Do you mean parameter `re` ?

Reply:
Yes, I'll fix.